GOSR2.py

```diff
@@ -0,0 +1,200 @@
+import os
+
+from Bio import SeqIO
+from Bio import SeqUtils
+from Bio import SeqRecord
+from abc import ABC, abstractmethod
+
+
+def filter_fastq(input_path: str, output_filename: str = '', gc_bounds: tuple = (0, 100), length_bounds: tuple = (0, 2**32), quality_threshold: int = 0)->list:
+    """
+    Filter fastq-sequences based on its gc-content, length and quality.
+    arguments:
+        - input_path (str): path to the fastq-file for reading
+        - output_filename (str): path for saving file with result (filtered fastq)
+        - gc_bounds (tuple): lower and upper borders for gc-content of sequence in percent
+        - length_bounds (tuple): lower and upper borders for length of sequence
+        - quality_threshold (int): lower border for quality of sequence
+    return:
+        - list of SeqRecords: filtered fastq-sequences
+    """
+    
+    if not isinstance(gc_bounds, tuple):
+        gc_bounds = tuple((0, int(gc_bounds)))
+    if not isinstance(length_bounds, tuple):
+        length_bounds = tuple((0, int(length_bounds)))
+
+    suitable_seqs = list(
+        record
+        for record in SeqIO.parse(input_path, "fastq")

```

Comment:
Может быть тут стоит добавить проверку файла, что он действительно нормальный?

```suggestion
    
if not os.path.isfile(input_path):
    raise FileNotFoundError(f"Input file '{input_path}' not found.")

suitable_seqs = list(
    record
    for record in SeqIO.parse(input_path, "fastq") 
    if ((gc_bounds[0] <= SeqUtils.gc_fraction(record.seq)*100 <= gc_bounds[1]) and 
        (length_bounds[0] <= len(record) <= length_bounds[1]) and 
        (sum(record.letter_annotations['phred_quality']) / len(record) >= quality_threshold))
)

```